docs/customization/controller.rst

```diff
@@ -122,6 +115,44 @@ getting a list of recommended products from your external api.
             arguments: ['@sylius.repository.product']
             public: true
 
+
+.. note::
+
+    Example of product provider with faked information
+
+    .. code-block:: yaml
+
+        # config/services.yaml
+        services:
+            app.provider.product:
+                class: App\Provider\ProductProvider
+                arguments: ['@sylius.fixture.example_factory.product']
+                public: true
+
+    .. code-block:: php
+
+        <?php
+
+        namespace App\Provider;
+
+        use Sylius\Bundle\CoreBundle\Fixture\Factory\ProductExampleFactory;
+        use Sylius\Component\Core\Model\ProductInterface;
+
+        class ProductProvider
+        {
+            /** @var ProductExampleFactory */
+            private $productExampleFactory;
+
+            public function __construct(ProductExampleFactory $productExampleFactory)
+            {
+                $this->productExampleFactory = $productExampleFactory;
+            }
+            public function getRecommendedProducts(): ProductInterface
+            {
+                return $this->productExampleFactory->create();
+            }
+        }
```

Comment:
if we decided to add example of this provider i suggest to make it really dummy:
```suggestion
            public function getRecommendedProducts(): array
            {
                // some custom logic with getting recommended products
                return $recommenedProducts;
            }
        }
```

but I agree with @GSadee that it is not necessary and helpful :(